### PR TITLE
Algolia search index update

### PR DESCRIPTION
**Pull request by VladyslavBondarenko**
Update search index on deploy using [algolia-indexing](https://www.npmjs.com/package/algolia-indexing).

Addition to https://github.com/codemagic-ci-cd/codemagic-docs/pull/358

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{
+    "name": "codemagic-docs",
+    "version": "1.0.0",
+    "scripts": {
+        "algolia": "node algolia.js"
+    },
+    "dependencies": {
```
Comment (mikhail-tokarev) [inline package.json:7]: probably dev dependencies should be ok

File: .gitignore [hunk 2]
```diff
@@ -1,3 +1,4 @@
 /public/
 /resources/
 .DS_Store
+*node_modules/
```
Comment (mikhail-tokarev) [inline .gitignore:4]: add full path

File: .github/workflows/deploy.yml [hunk 3]
```diff
@@ -18,3 +18,7 @@ jobs:
       - run: curl --location https://github.com/gohugoio/hugo/releases/download/v0.77.0/hugo_extended_0.77.0_Linux-64bit.tar.gz | tar -vxzO hugo > hugo && chmod a+x hugo
       - run: ./hugo --environment production --minify
       - run: ./hugo deploy --environment production --invalidateCDN true --maxDeletes -1
+
+      - run: yarn install
```
Comment (mikhail-tokarev) [inline .github/workflows/deploy.yml:22]: should it be before `./hugo deploy`?

**Review by mikhail-tokarev — COMMENTED** [review]
added some comments

File: .gitignore (hunk 2 above)
Comment (mikhail-tokarev) [reply .gitignore:4]: should be only one node_modules otherwise there is some problem

File: .github/workflows/deploy.yml (hunk 3 above)
Comment (VladyslavBondarenko) [reply .github/workflows/deploy.yml:22]: do you mean only `yarn install` or together with `yarn run algolia`?

**Review by mikhail-tokarev — APPROVED** [review]
LGTM

**Pull request merged**